pilot/cmd/pilot-agent/status/util/stats.go

```diff
@@ -24,45 +24,66 @@ import (
 )
 
 const (
-	statCdsUpdatesSuccess   = "cluster_manager.cds.update_success"
-	statCdsUpdatesRejection = "cluster_manager.cds.update_rejected"
-	statLdsUpdatesSuccess   = "listener_manager.lds.update_success"
-	statLdsUpdatesRejection = "listener_manager.lds.update_rejected"
+	statCdsVersion    = "cluster_manager.cds.version"
+	statLdsVersion    = "listener_manager.lds.version"
+	statServerState   = "server.state"
+	versionStatsRegex = "^(cluster_manager.cds|listener_manager.lds).version$"
 )
 
+type stat struct {
+	name  string
+	value *uint64
+	found bool
+}
+
 // Stats contains values of interest from a poll of Envoy stats.
 type Stats struct {
-	CDSUpdatesSuccess   uint64
-	CDSUpdatesRejection uint64
-	LDSUpdatesSuccess   uint64
-	LDSUpdatesRejection uint64
+	// Hash of the contents from the last successful cluster update.
+	CDSVersion uint64
```

Comment:
The idea of previous metric is progress. It provides debug logs because the numbers shows progress, even it's not ready yet. 

@howardjohn WDYT?

Reply:
But the numbers are misleading in hot restart case. It shows 30 success and 10 rejected in hot restart case, even though it did not receive any updates/any updates rejected. It carries forward those values from previous envoy instance.  One thing I can change the output to more readable (like cds status : Received/Not Received, lds status Received/Not Received) rather than printing version (which is of not that much help)

Reply:
Thank you for the explanation!